rewrite-maven/src/main/java/org/openrewrite/maven/UpdateMavenModel.java

```diff
@@ -38,15 +38,16 @@ public Xml visitDocument(Xml.Document document, P p) {
 
         MavenResolutionResult resolutionResult = getResolutionResult();
         Pom requested = resolutionResult.getPom().getRequested();
+        requested.getProperties().clear();
 
         Optional<Xml.Tag> properties = document.getRoot().getChild("properties");
-        requested.getProperties().clear();
         if (properties.isPresent()) {
             for (final Xml.Tag propertyTag : properties.get().getChildren()) {
                 requested.getProperties().put(propertyTag.getName(),
                         propertyTag.getValue().orElse(""));
             }
         }
+        requested.getProperties().putAll(resolutionResult.getUserProperties());
```

Comment:
This will be `null` on old LSTs deserializing even though the field is non-null, so you should do a null check on this here.